contracts/BalanceQuery.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "./libraries/Authorizable.sol";
+import "./interfaces/ICoreVoting.sol";
+import "./interfaces/IVotingVault.sol";
+
+contract BalanceQuery is Authorizable {
+    // stores approved voting vaults
+    IVotingVault[] public vaults;
+
+    /// @notice Constructs this contract and stores needed data
+    /// @param _owner The contract owner authorized to remove vaults
+    constructor(address _owner, address[] memory votingVaults) {
+        // set the storage array of vaults
+        vaults = new IVotingVault[](votingVaults.length);
```

Comment:
This line sets vaults to a new array of 0s which is the same length as `votingVaults`, I think the right syntax is actually `vaults = votingVaults;`